tests/test.lua

```diff
@@ -212,6 +215,10 @@ local cjson_tests = {
       json.encode, { 10 }, true, { '10' } },
     { "Encode string",
       json.encode, { "hello" }, true, { '"hello"' } },
+    { "Encode empty table as array",
+      json.encode, { empty_array }, true, { '[]' } },
+    { "Encode empty table as table",
+      json.encode, { {} }, true, { '{}' } },
```

Comment:
Got it.  I am going to send another refined pull-request later around.

On Fri, Jul 17, 2015 at 1:07 PM Yichun Zhang <notifications@github.com>
wrote:

> In tests/test.lua
> <https://github.com/openresty/lua-cjson/pull/1#discussion_r34862504>:
>
> > @@ -212,6 +215,10 @@ local cjson_tests = {
> >        json.encode, { 10 }, true, { '10' } },
> >      { "Encode string",
> >        json.encode, { "hello" }, true, { '"hello"' } },
> > +    { "Encode empty table as array",
> > +      json.encode, { empty_array }, true, { '[]' } },
> > +    { "Encode empty table as table",
> > +      json.encode, { {} }, true, { '{}' } },
>
> I suggest adding some more complicates cases here. Your last example can
> be a good candidate, for example.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openresty/lua-cjson/pull/1/files#r34862504>.
>